gobblin-utility/src/main/java/org/apache/gobblin/util/filesystem/FileSystemLimiterKey.java

```diff
@@ -30,10 +30,16 @@
 
   public static final String RESOURCE_LIMITED_PREFIX = "filesystem";
   private final URI uri;
+  public final String serviceName;
 
   public FileSystemLimiterKey(URI uri) {
-    super(RESOURCE_LIMITED_PREFIX + "/" + getFSIdentifier(uri));
+    this(uri, null);
+  }
+
+  public FileSystemLimiterKey(URI uri, String serviceName) {
+    super(RESOURCE_LIMITED_PREFIX + "/" + getFSIdentifier(uri) + serviceName == null ? "" : "/" + serviceName);
```

Comment:
can you add parentheses to remove ambiguity in operator precedence? `+ (serviceName == null ? "" : "/" + serviceName)`. Also, please also check whether the service name is empty, you can use `StringUtils.isNullOrEmpty`